app/assets/javascripts/school_administrator_dashboard/discipline_dashboard/SchoolDisciplineDashboard.js

```diff
@@ -96,13 +161,33 @@ export default class SchoolDisciplineDashboard extends React.Component {
     });
   }
 
-  //For grades and classrooms, the students table should only show the relevant students
-  groupStudents() {
-    if (this.state.selectedChart === 'grade' && this.state.selectedCategory) {
-      return this.props.dashboardStudents.filter(student => student.grade === this.state.selectedCategory);
-    } else if (this.state.selectedChart === 'classroom' && this.state.selectedCategory) {
-      return this.props.dashboardStudents.filter(student => student.homeroom_label === this.state.selectedCategory);
-    } else return this.props.dashboardStudents;
+  sortedByStudent(groupedStudents) {
+    return Object.keys(groupedStudents).sort((a,b) => {
+      return this.getIncidentsFromStudents(groupedStudents[b]).length - this.getIncidentsFromStudents(groupedStudents[a]).length;
+    });
+  }
+
+  studentTableRows(students) {
+    const filteredStudents = this.filterStudents(students);
+    const {selectedCategory} = this.state;
+    return filteredStudents.map(student => {
+      return {
+        id: student.id,
+        first_name: student.first_name,
+        last_name: student.last_name,
+        grade: student.grade,
+        latest_note: student.latest_note,
+        events: this.filterIncidents(student.discipline_incidents, selectedCategory).length
+      };
+    });
+  }
+
+  allIncidentTypes(students) {
+    return _.uniqBy(this.getIncidentsFromStudents(students), 'incident_code').map(incident => incident.incident_code);
```

Comment:
This method is called `allIncidentTypes` but it looks like it's ultimately calling `filterIncidents`.  Is this a naming thing or is this a bug?

Reply:
Oops, that's a bug - I added the filter to the student incident counts later, but we don't want this list filtered.